base/broadcast.jl

```diff
@@ -1211,4 +1211,29 @@ end
 end
 @inline broadcasted(::S, f, args...) where S<:BroadcastStyle = Broadcasted{S}(f, args)
 
+function _lazy end  # only used in `@:` macro; does not have to be callable
```

Comment:
`_lazy <: Function` does not seem right. 

`struct _lazy end` and an exported `const Lazy = _lazy()` with `broadcasted(::_lazy, x) = (x,)` could be an alternative (users would write either `sum(@: a .* b)` or `sum(Base.Broadcast.instantiate(Lazy.( a.* b))))`.

Reply:
It does hold the place of a function, though.  Either which way, it's necessarily a bit of a strange construct, so I'm not sure it really matters.

I don't want to see it exported for exactly that reason.  It's odd (but necessary).